codegen/src/main/scala/bsp/codegen/docs/MarkdownRenderer.scala

```diff
@@ -25,69 +32,88 @@ class MarkdownRenderer private (tree: DocTree, visited: MSet[ShapeId]) {
 
   def render: Lines = {
     val commonShapes =
-      if (tree.commonShapes.isEmpty) empty
+      if (tree.commonShapeIds.isEmpty) empty
       else
         lines(
           "## Common shapes",
           newline,
-          tree.commonShapes.foldMap(renderNode),
+          tree.commonShapeIds.foldMap(renderStructureNode),
           newline
         )
 
-    val services = tree.services.foldMap(renderNode)
-    commonShapes ++ services
+    // Server comes first because it's bigger.
+    val services = tree.services.sortBy(-_.operations.size).foldMap(renderServiceNode)
```

Comment:
I find `-_.` a bit funky here. Wouldn't
```suggestion
    val services = tree.services.sortBy(_.operations.size).reverse.foldMap(renderServiceNode)
```
Be the same and maybe easier to read when glancing over this?